File: dpo_train/generation_output/cpp-py/HumanEval_108_count_nums.py
### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
long count_nums(std::vector<long> arr) {
    long count = 0;
    for (int i = 0; i < arr.size(); i++) {
        long n = arr[i];
        long neg = 1;
        if (n < 0) {
            n = -1 * n;
            neg = -1;
        }
        std::vector<long> digits;
        std::stringstream ss;
        ss << n;
        std::string str = ss.str();
        for (int j = 0; j < str.size(); j++) {
            digits.push_back(str[j] - '0');
        }
        digits[0] = digits[0] * neg;
        long sum = 0;
        for (int j = 0; j < digits.size(); j++) {
            sum += digits[j];
        }
        if (sum > 0) {
            count++;
        }
    }
    return count;
}

### Python version

from typing import List

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### C++ version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

### Python version

def count_nums(arr: List[int]) -> int:
    return count_nums(arr[0])

File: dpo_train/generation_output/py-cpp/HumanEval_108_count_nums.cpp
### Python version

from typing import List

def count_nums(arr: List[int]) -> int:
    def digits_sum(n):
        neg = 1
        if n < 0: n, neg = -1 * n, -1 
        n = [int(i) for i in str(n)]
        n[0] = n[0] * neg
        return sum(n)
    return len(list(filter(lambda x: x > 0, [digits_sum(i) for i in arr])))

### C++ version

#include<assert.h>
#include<bits/stdc++.h>
#include<vector>
#include<sstream>
#include<regex>
#include<map>
#include<set>
#include<unordered_map>
#include<unordered_set>
long count_nums(std::vector<long> arr) {
    long res = 0;
    for (long i = 0; i < arr.size(); i++) {
        res += count_nums(arr[i]);
    }
    return res;
}

int main() {
    std::vector<long> arr = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
    std::cout << count_nums(arr) << std::endl;
    return 0;
}
